src/network/webtask.cpp

```diff
@@ -123,32 +123,24 @@ void WebTask::onNetworkError(
         QNetworkReply::NetworkError errorCode,
         const QString& errorString,
         const WebResponseWithContent& responseWithContent) {
-    DEBUG_ASSERT(m_state == State::Pending);
+    DEBUG_ASSERT(m_state == State::Failed || m_state == State::Pending);
```

Comment:
How could the state be already entered failed when a network reply arrives?

Reply:
This is because this function is also called, before the request is issued and the state goes to the pending state.
https://github.com/mixxxdj/mixxx/blob/41abb4da5f68b796db44a3b52f8903115c3bcd3b/src/network/webtask.cpp#L200
Before, the state was artificially set to State::Pending, even though there was already a failure detected.
Probably just to trick the original assertion.
This is fixed now for a traceable usage of states.